src/vivarium_testing_utils/automated_validation/comparison.py

```diff
@@ -222,25 +226,36 @@ def _get_metadata_from_dataset(
 
     def _align_datasets(self) -> tuple[pd.DataFrame, pd.DataFrame]:
         """Resolve any index mismatches between the test and reference datasets."""
-        test_data = self.test_data.copy()
-        reference_data = self.reference_data.copy()
-        # If the test data has any index levels that are not in the reference data, marginalize
-        # over those index levels.
+        # Get union of test data index names
+        combined_test_index_names = set(
+            index_name
+            for key in self.test_datasets
+            for index_name in self.test_datasets[key].index.names
+        )
+        # Get index levels that are only in the test data.
         test_only_indexes = [
             index
-            for index in self.test_data.index.names
+            for index in combined_test_index_names
             if index not in self.reference_data.index.names
         ]
-        stratified_test_data = marginalize(test_data, test_only_indexes)
-
-        # Drop any singular index levels from the reference data if they are not in the test data.
-        # If any ref-only index level is not singular, raise an error.
+        # Likewise, get index levels that are only in the reference data.
         ref_only_indexes = [
             index
             for index in self.reference_data.index.names
-            if index not in self.test_data.index.names
+            if index not in combined_test_index_names
         ]
+
+        # If the test data has any index levels that are not in the reference data, marginalize
+        # over those index levels.
+        test_datasets = self.test_datasets.copy()
+        test_datasets = {
```

Comment:
Nit/question: Why do you need to define the copy then overwrite it? Can you just pass the original or copy into the function?